### ISSUE TITLE
Documentations Improvements

### ISSUE THREAD

**Issue by rlugge** [ZipArchive/ZipArchive#416]
So the existing documentation is extremely scarce.  The readme file's data is extremely limited, exposing only the very most basic bare bones of the module's API.  It's confusing, and doesn't explain half of what can be done with the basic interfaces.

If there aren't any current plans to improve documentation, please provide guidance on how you'd prefer to get it added -- either via additional wiki articles or by updating the readme via a fork/PR cycle, and I'd be glad to do the work (to the best of my ability anyways, since I'm sure even my understanding of the interface falls short of complete)

**Comment by Coeur** [comment]
I prefer pull requests of the current Readme file.

Wiki, in my opinion, is only to be used for describing old versions of the lib, or how to migrate from an old version to a newer one.

**Comment by jhudsonWA** [comment]
Agreed. Would love some pull requests in this area.

**Comment by jhudsonWA** [comment]
@rlugge  Any update on this? Is this PR still something you are willing to work on?

**Comment by rlugge** [comment]
@jhudsonWA yes.  Sorry, I've been busy with work and haven't had free time to work on this yet.  Project started running behind and I've been trying to catch up.

**Comment by jhudsonWA** [comment]
@rlugge No worries : ) Not rushing you, just wanted to see if I should leave it open or not. Story of my life. @Coeur  has been waiting on me for months. :P

**Comment by Coeur** [comment]
@rlugge still up to help us with the documentation? :)

**Comment by rlugge** [comment]
Yes, thanks for the reminder.
